shared/chat/conversation/container.js

```diff
@@ -155,5 +172,20 @@ export default compose(
     onMouseOverSearchResult: props => id => {
       props.onUpdateSelectedSearchResult(id)
     },
+  }),
+  lifecycle({
+    componentWillUnmount: function() {
+      this.props.onStoreInputText(this.props.chatText)
+    },
+    componentWillReceiveProps: function(nextProps) {
+      if (
+        this.props.selectedConversationIDKey &&
+        this.props.selectedConversationIDKey !== nextProps.selectedConversationIDKey
+      ) {
+        this.props.onStoreInputText(this.props.chatText)
+        // withState won't get called again if props changes!
```

Comment:
yeah, this comment wasn't me. I moved this code from here: https://github.com/keybase/client/pull/7909/files/6b4e34e6962a7fa4d1f461e69969e278d8eb32a5..bcf488eab01d5f47185146981041b1af3192fa83#diff-d26537c6ed305323863d47cef8793f3cL140

We can get rid of the comment since it just describes the behavior of withState